src/Symfony/Component/Notifier/Bridge/Telegram/TelegramTransportFactory.php

```diff
@@ -42,4 +43,17 @@ protected function getSupportedSchemes(): array
     {
         return ['telegram'];
     }
+
+    private function getToken(Dsn $dsn): string
+    {
+        if (null === $dsn->getUser() && null === $dsn->getPassword()) {
+            throw new IncompleteDsnException('Missing token');
+        }
+
+        if (null === $dsn->getPassword()) {
+            throw new IncompleteDsnException('Malformed token');
```

Comment:
Same here. Instead of `Malformed token`, we could say `The given DSN is missing the password.`

Reply:
This exception is thrown when the configured token does not contain a colon. So in this case the token is indeed malformed. This may seem confusing when looking at it in the code, but for the developer the error message should be more helpful as it based on the terminology of the Telegram api.

Reply:
Yes, after reading your explanation it makes total sense. Thanks.